eng/tools/oav-runner/src/runner.ts

```diff
@@ -11,20 +11,30 @@ import {
 } from "@azure-tools/specs-shared/changed-files"; //getChangedFiles,
 import { ReportableOavError } from "./formatting.js";
 
-export async function checkExamples(
-  rootDirectory: string,
-): Promise<[number, string[], ReportableOavError[]]> {
-  let errors: ReportableOavError[] = [];
-
-  const changedFiles = await getChangedFiles({
-    cwd: rootDirectory,
-  });
+export async function preCheckFiltering(rootDirectory: string, fileList?: string[]): Promise<string[]> {
+  const changedFiles = fileList?.length
+    ? fileList
+    : await getChangedFiles({ cwd: rootDirectory });
```

Comment:
```suggestion
  const changedFiles = fileList ?? await getChangedFiles({ cwd: rootDirectory });
```

Shorter and I think sufficient, unless it's important to special-case empty array.